src/uct/tcp/tcp_ep.c

```diff
@@ -461,28 +562,55 @@ ucs_status_t uct_tcp_ep_am_short(uct_ep_h uct_ep, uint8_t am_id, uint64_t header
 {
     uct_tcp_ep_t *ep       = ucs_derived_of(uct_ep, uct_tcp_ep_t);
     uct_tcp_iface_t *iface = ucs_derived_of(uct_ep->iface, uct_tcp_iface_t);
+    ucs_status_t status;
     uct_tcp_am_hdr_t *hdr;
 
-    UCT_TCP_AM_PREPARE(ep, am_id, iface->config.short_size - sizeof(*hdr),
-                       hdr, memcpy, payload, length, header, SHORT, "am_short");
+    status = uct_tcp_ep_am_check_length(length + sizeof(header),
+                                        iface->config.short_size - sizeof(uct_tcp_am_hdr_t),
+                                        "am_bcopy");
+    if (ucs_unlikely(status != UCS_OK)) {
+        UCS_STATS_UPDATE_COUNTER(ep->super.stats, UCT_EP_STAT_NO_RES, 1);
```

Comment:
no need to increase counter here, it's wrong parameter case, not a resource outage

Reply:
thanks. fixed